### PR TITLE
Introduce Task class to store information about the task and get rid of RawFormat.

### PR DESCRIPTION
**Pull request by ehlemur-zz**

### CODE REVIEW

**Comment by ehlemur-zz** [comment]
@pbos @kwiberg 
Please take a look

File: gtest-parallel [hunk 1]
```diff
@@ -322,8 +353,6 @@ parser.add_option('--gtest_filter', type='string', default='',
                   help='test filter')
 parser.add_option('--gtest_also_run_disabled_tests', action='store_true',
                   default=False, help='run disabled tests too')
-parser.add_option('--format', type='string', default='filter',
-                  help='output format (raw,filter)')
```
Comment (kwiberg) [inline gtest-parallel:326]: Removing this flag is a behavior change. Why is it in this patch, and are there any other behavior changes?

**Review by kwiberg — COMMENTED** [review]
This certainly looks like a readability improvement. I'm interested to hear why this isn't a pure refactoring, though.

**Comment by ehlemur-zz** [comment]
We discussed it with Peter and decided to get rid of RawFormat and add a
Task class to store information about each execution.
There shouldn't be any other behavior changes.

Den 21 apr. 2017 03:37 skrev "Karl Wiberg" <notifications@github.com>:

*@kwiberg* commented on this pull request.

This certainly looks like a readability improvement. I'm interested to hear
why this isn't a pure refactoring, though.
------------------------------

In gtest-parallel
<https://github.com/google/gtest-parallel/pull/23#discussion_r112593865>:

> @@ -322,8 +353,6 @@ parser.add_option('--gtest_filter', type='string', default='',
                   help='test filter')
 parser.add_option('--gtest_also_run_disabled_tests', action='store_true',
                   default=False, help='run disabled tests too')
-parser.add_option('--format', type='string', default='filter',
-                  help='output format (raw,filter)')

Removing this flag is a behavior change. Why is it in this patch, and are
there any other behavior changes?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/google/gtest-parallel/pull/23#pullrequestreview-33921618>,
or mute the thread

**Review by kwiberg — APPROVED** [review]
OK, that's fine then.

**Comment by ehlemur-zz** [comment]
Friendly ping.

**Pull request merged**